etcd/KeepAlive.hpp

```diff
@@ -36,6 +52,13 @@ namespace etcd
      */
     void Cancel();
 
+    /**
+     * Check if the keep alive is still valid (invalid when there's an async exception).
```

Comment:
@sighingnow I think here we also need to check if the last message from the server was with a timeout of -1 - I think this also indicates that the keep-alive is no longer valid. This can of course be checked with existing methods.

Reply:
Agree. Will commit such an enhancement later.

However please note that when constructing the `KeepAlive` object the `ttl` argument should be less than the TTL that used to grant the lease, and inside the implementation of `KeepAlive` the refresh the lease every `ttl - 1` seconds.

Thus generally this shouldn't be a problem, unless we encounter some problems on the server, or the lease is cancelled in an external thread.

Reply:
Thanks!